process/transaction/transactionCostEstimator.go

```diff
@@ -151,17 +168,22 @@ func (tce *transactionCostEstimator) simulateTransactionCost(tx *transaction.Tra
 	}, nil
 }
 
-func computeGasUnitsBasedOnVMOutput(tx *transaction.Transaction, vmOutput *vmcommon.VMOutput) uint64 {
+func (tce *transactionCostEstimator) computeGasUnitsBasedOnVMOutput(tx *transaction.Transaction, vmOutput *vmcommon.VMOutput) uint64 {
 	isTooMuchGasProvided := strings.Contains(vmOutput.ReturnMessage, smartContract.TooMuchGasProvidedMessage)
 	if !isTooMuchGasProvided {
 		return tx.GasLimit - vmOutput.GasRemaining
 	}
 
-	return tx.GasLimit - extractGasRemainedFromMessage(vmOutput.ReturnMessage)
+	if tce.flagTooMuchGasV2Msg.IsSet() {
+		gasNeededForProcessing := extractGasRemainedFromMessage(vmOutput.ReturnMessage, gasUsedSlitString)
```

Comment:
interesting as those new lines are not covered by tests

Reply:
TestAsyncESDTTransfer is getting here.